api/src/main/java/org/arquillian/cube/ContainerObjectFactory.java

```diff
@@ -0,0 +1,32 @@
+package org.arquillian.cube;
+
+/**
+ * Factory to instantiate container objects
+ *
+ * @author <a href="mailto:rivasdiaz@gmail.com">Ramon Rivas</a>
+ */
+public interface ContainerObjectFactory {
+
+    /**
+     * Creates an instance of the container object. It also creates and starts a cube defined by the container object
+     * class. Some configuration can be overridden by passing an additional ContainerObjectConfiguration instance.
+     *
+     * @param containerObjectClass type of the container object to instantiate. Required
+     * @param configuration if not null, allows specifying some configuration parameters. Optional (can be null)
```

Comment:
I have noticed this now, I don't like that users can add `null` value in public API. Of course there are some exceptions, but I usually prefer in public APIs to create a `Nullified` object or just create a new method. So for example:

`<T> T createContainerObject(Class<T> containerObjectClass)` and then if you want internally you call `this.createContainerObject(containerObjectClass, null)`

Another approach might be the nullified object which `ContainerObjectConfiguration` class has an static method called `defaults()` which returns an empty object and then inside the method again you can do something like:

```
if (configuration == ContainerObjectConfiguration.defaults()) {
  this.createContainerObject(containerObjectClass, null)
}
```

What I am trying to do is that the API does not expose the null in public API.